### DISCUSSION TITLE
requiredIf Validation not working

### DISCUSSION THREAD

**Discussion by oneplusone-solutions** [filamentphp/filament#6774] [Help] [answered]
![image](https://github.com/filamentphp/filament/assets/63426914/08d081df-6faa-44e4-b3f7-1b1612b07399)
![image](https://github.com/filamentphp/filament/assets/63426914/bdda0cd5-27d6-4e3c-9c3d-d852bbebe7b4)

Have a problem with validation - ->requiredIf('text', null) - return an error
 ->rule( 'required_if:title,null' ) - not working, always return success

Please help me to do corect validation(

**Comment by pboivin** [comment]
Maybe this could work : 
```php
    ->required(fn ($get) => ! $get('text'))
```

**Reply by GalinaBublik** [reply]
It work always us required( 
![image](https://github.com/filamentphp/filament/assets/10061993/a2568669-9e4d-45bf-ad2e-2203d5ec7de3)

**Reply by danharrin** [reply]
Make the title `->lazy()`

**Reply by GalinaBublik** [reply]
->lazy() didn't help(

**Reply by danharrin** [reply]
sorry, I meant the `text` or whatever field is inside `$get()`

**Reply by GalinaBublik** [reply, marked as answer]
```
TextInput::make('title')
                        ->label('Title')
                        ->default(null)
                        ->required(fn ($get) => ! $get('text'))
                        ->lazy()
                        ->dehydrated(fn (Page $livewire) => $livewire instanceof CreateRecord)
                        ->maxLength(50),
                    RichEditor::make('text')
                        ->lazy()
                        ->label('Description')
                        ->default(null)
                        ->required(fn ($get) => ! $get('title'))
                        ->maxLength(300)
                        ->disableToolbarButtons([
                            'attachFiles',
                        ]),
```

Look like it works as necessary - Required title OR text field.

Thank a lot!
